### ISSUE TITLE
iSCSI based SBD

### ISSUE THREAD

**Issue by weikiat8** [SUSE/ha-sap-terraform-deployments#482]
is iSCSI target based SBD in this deployment better than using cloud-native STONITH agents?

For example:
AWS - Overlay IP-based STONITH: external/ec2
Azure - Fencing Agent-based STONITH: fence_azure_arm 
GCP - external/gcpstonith
Ali Cloud: fence_aliyun

For SAP deployments typically stability is key, but what is the technicalities and rationale behind NOT using cloud-native Fencing agent?
In Azure, Microsoft cautioned the downside of using its own fence agent, besides the longer failover what would be other reasons?

“If you don't want to invest in one additional virtual machine, you can also use the Azure Fence agent. The downside is that a failover can take between 10 to 15 minutes if a resource stop fails or the cluster nodes cannot communicate which each other anymore.”

**Comment by diegoakechi** [comment]
Hi @weikiat8,

I will not say what is better or not, because all of them have pros and cons.

SBD is a platform agnostic fencing mechanism, so it could be an advantage to use it in case you have hybrid environment, like running cluster on premises and in the cloud, or in different clouds, so what could help in terms of learning curve and adminstration since you could have the same standard on all clusters. On the other side, it comes with the overhead of administering an iSCSI target extra machine as there is (not yet) shared block devices on the Cloud Providers. 

The Cloud Providers native fence agents uses specifics APIs for each provider and in some of them there is this limitation of long reaction times (AFAIK improvements on that regards are being done in all of them). On the other side they are actively maintained, tested and certified by the providers, so improvements are always being introduced, and you save the costs of extra machines and disks only for the fencing mechanism.

So basically you have choices and need to pick what fits better for your environment.

Hope it answers your question.

**Comment by weikiat8** [comment]
thanks for the detailed explanation! Appreciate it
